mediapipe/tasks/ios/genai/core/sources/LlmTaskRunnerRefactored.swift

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksGenAIC
+
+/// This class is used to create and call appropriate methods on the C `LlmInferenceEngine` to
+/// initialize an LLMInference task and create prompt sessions.
+final class LlmTaskRunner {
```

Comment:
@yishuangP Can we rename this to `LlmEngineRunner` to keep it closer to the C APIs? Not sure how Java will be structured.